File: servers/jax-rs/src/main/java/com/wordnik/api/ChatApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.client.model.Chat;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/chat.json")
@Api(value = "/chat", description = "the chat API")
@Produces({"application/json"})
public class ChatApi {
  @GET
  @Path("/")
  @ApiOperation(value = "Get chat messages.", notes = "", responseClass = "List<Chat>")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response get(
    @ApiParam(value = "Starting point for results."
    ,required=true
)@QueryParam("start")
 Double start
    ,@ApiParam(value = "Number of results to fetch."
    ,required=true
, defaultValue="100"
)@QueryParam("count")
 Double count
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  @POST
  @Path("/")
  @ApiOperation(value = "Send a chat message.", notes = "", responseClass = "Chat")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response send(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  }

File: servers/scalatra/src/main/scala/com/wordnik/client/api/ChatApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.Chat
import com.wordnik.client.model.Error
import java.math.BigDecimal
import com.wordnik.client.model.ConnectedUsers

import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class ChatApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "ChatApi"
  override protected val applicationName: Option[String] = Some("Chat")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }
  

  val chat.getOperation = (apiOperation[List[Chat]]("chat.get")
      summary "Get chat messages."
      parameters(queryParam[BigDecimal]("count").description("").optional.defaultValue(100),
        queryParam[BigDecimal]("start").description("").optional.defaultValue(0),
        queryParam[Boolean]("reverse").description("").optional.defaultValue(true))
  )

  get("/chat",operation(chat.getOperation)) {
    
    
    
        
      
      val count = params.getAs[BigDecimal]("count")
            

    
    println("count: " + count)
  
    
    
        
      
      val start = params.getAs[BigDecimal]("start")
            

    
    println("start: " + start)
  
    
    
        
      
      val reverse = params.getAs[Boolean]("reverse")
            

    
    println("reverse: " + reverse)
  
  }

  

  val chat.sendOperation = (apiOperation[Chat]("chat.send")
      summary "Send a chat message."
      parameters(formParam[String]("message").description(""))
  )

  post("/chat",operation(chat.sendOperation)) {
    
    
    
                
      val message = params.getAs[String]("message")
    

    
    println("message: " + message)
  
  }

  

  val chat.getConnectedOperation = (apiOperation[ConnectedUsers]("chat.getConnected")
      summary "Get connected users."
      parameters()
  )

  get("/chat/connected",operation(chat.getConnectedOperation)) {
    
  }

}